### PR TITLE
improve(core): update voting upgrade script to pass proposer wallet as migrated address

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

To allow the foundation hot wallet to claim rewards on behalf of voters in the old voting contract, we need to set it as the migrated address.

**Summary**

Added a parameter to the upgrader contract and passed the foundation hot wallet to the constructor in the upgrade script.

Note: the migrated address has the following abilities:
- Migrated address can request a price to "start a vote"
- Nobody can cast votes (so any incomplete votes cannot be finalized and, therefore generate rewards).
- Migrated address can call claim rewards.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

File: packages/core/scripts/voting-upgrade-umip/3_Verify.js [hunk 1]
```diff
@@ -37,7 +38,7 @@ async function runExport() {
 
   const newVotingContract = await Voting.at(votingAddress);
   const currentOwner = await newVotingContract.owner();
-  assert.equal(web3.utils.toChecksumAddress(currentOwner), web3.utils.toChecksumAddress(Governor.address));
+  assert.equal(web3.utils.toChecksumAddress(currentOwner), web3.utils.toChecksumAddress(governor.address));
```
Comment (chrismaree) [inline packages/core/scripts/voting-upgrade-umip/3_Verify.js:41]: curious question: why was this change required? if you are running this with truffle can you not pull the address directly from the truffle artefact?

File: packages/core/contracts/umips/VotingUpgrader.sol [hunk 2]
```diff
@@ -26,23 +26,31 @@ contract VotingUpgrader {
...
+     *                            old voting contract (used to claim rewards on others' behalf). Note: this address
+     *                            can always be changed by the voters.
      */
     constructor(
         address _governor,
         address _existingVoting,
         address _newVoting,
-        address _finder
+        address _finder,
+        address _setMigratedAddress
```
Comment (chrismaree) [inline packages/core/contracts/umips/VotingUpgrader.sol:47]: you could just use `msg.sender` for this, no? I guess it's more generalised in how you've done it now and in future we might want to set the migrated address to the new voting contract so not really any downside of doing it this way.

**Review by chrismaree — APPROVED** [review]

File: packages/core/scripts/voting-upgrade-umip/3_Verify.js (hunk 1 above)
Comment (mrice32) [reply packages/core/scripts/voting-upgrade-umip/3_Verify.js:41]: I think this broke when we moved over to using our own `getTruffleArtifact` method. `.address` is seeded by some async method being called on the instance so it's able to grab the network id from web3. Truffle does this for it's artifacts on startup, but since this hadn't been done yet for the artifact that we created, the instance doesn't know what address to grab.

File: packages/core/contracts/umips/VotingUpgrader.sol (hunk 2 above)
Comment (mrice32) [reply packages/core/contracts/umips/VotingUpgrader.sol:47]: Good point. We could. But yeah, I thought it might be better to keep it explicit and agnostic of the deployer.

**Review by nicholaspai — APPROVED** [review]
LGTM
